SerialPrograms/Source/PokemonLA/Programs/General/PokemonLA_RamanasIslandCombee.cpp

```diff
@@ -0,0 +1,315 @@
+    return std::unique_ptr<StatsTracker>(new Stats());
+}
+
+void RamanasCombeeFinder::check_tree_no_stop(SingleSwitchProgramEnvironment& env, BotBaseContext& context){
+    context.wait_for_all_requests();
+    disable_shiny_sound(context);
+    // Throw pokemon
+    pbf_press_button(context, BUTTON_ZR, (0.5 * TICKS_PER_SECOND), 1.5 * TICKS_PER_SECOND);
+    context.wait_for_all_requests();
+    env.current_stats<Stats>().trees++;
+    env.update_stats();
+}
+
+bool RamanasCombeeFinder::check_tree(SingleSwitchProgramEnvironment& env, BotBaseContext& context){
+    context.wait_for_all_requests();
+
+    disable_shiny_sound(context);
+    bool battle_found = check_tree_or_ore_for_battle(env.console, context);
+    env.current_stats<Stats>().trees++;
+    env.update_stats();
+
+    context.wait_for_all_requests();
+
+    bool ret = false;
+    if (battle_found){
+        ret = handle_battle(env, context);
+    }
+
+    enable_shiny_sound(context);
+
+    return ret;
+}
+
+
+void RamanasCombeeFinder::disable_shiny_sound(BotBaseContext& context){
+    context.wait_for_all_requests();
+    m_enable_shiny_sound.store(false, std::memory_order_release);
+}
+void RamanasCombeeFinder::enable_shiny_sound(BotBaseContext& context){
+    context.wait_for_all_requests();
+    m_enable_shiny_sound.store(true, std::memory_order_release);
+}
+
+bool RamanasCombeeFinder::handle_battle(SingleSwitchProgramEnvironment& env, BotBaseContext& context){
+    Stats& stats = env.current_stats<Stats>();
+
+    PokemonDetails pokemon = get_pokemon_details(env.console, context, LANGUAGE);
+
+    pbf_press_button(context, BUTTON_B, 20, 225);
+
+    context.wait_for_all_requests();
+
+    if (pokemon.name_candidates.find("combee") == pokemon.name_candidates.end()){
+        env.console.log("Not a Combee, leaving battle.");
+        exit_battle(env.console, context);
+        return false;
+    }
+
+    env.console.log("Combee found");
+    stats.found++;
+    on_match_found(env, env.console, context, SHINY_DETECTED_ENROUTE, true);
+    return true;
+}
+
+void RamanasCombeeFinder::grouped_path(SingleSwitchProgramEnvironment& env, BotBaseContext& context){
+
+    BattleMenuDetector battle_menu_detector(env.console, env.console, true);
+
+    int ret = run_until(
+        env.console, context,
+        [&](BotBaseContext& context){
+
+            env.console.log("Checking Tree 1");
+            change_mount(env.console,context,MountState::BRAVIARY_ON);
+            pbf_move_left_joystick(context, 241, 0, 100, 20);
+            pbf_press_button(context, BUTTON_B, 2390, 0);
+            pbf_press_button(context, BUTTON_Y, 380, 0);
+            pbf_move_right_joystick(context, 127, 255, 90, 20);
+            check_tree_no_stop(env, context);
+
+            env.console.log("Checking Tree 2");
+            pbf_press_button(context, BUTTON_PLUS, 20, 200);
+            pbf_move_left_joystick(context, 242, 0, 100, 20);
+            enable_shiny_sound(context);
```

Comment:
Since the button presses are async, you can't reliably schedule the enable-shiny-sound here. You'll need to precede it with a wait-for-all. However, once you do that, the shiny sound won't turn back on until after you've already started moving again - thus creating a window to missing an enroute shiny. So you'll actually want to move it up one line.

```
pbf_press_button(context, BUTTON_PLUS, 20, 200);
context.wait_for_all_requests();
enable_shiny_sound(context);
pbf_move_left_joystick(context, 242, 0, 100, 20);
```

Reply:
got it! 
done!